### DISCUSSION TITLE
getLEDs not passing right?

### DISCUSSION THREAD

**Discussion by Tempy111** [ArminJo/DigistumpArduino#44] [Q&A] [answered]
I don't think this is a bug persay so i'm putting it here but..

Trying a bit of basic code to detect the Caps lock and num lock status and light up one of two LEDS based on it.. my Ardunio version works, trying to change it for the digispark ATTiny and.. well.. I think i need to change the pins i'm using.. I put one LED with the positive on Pin 0 and on Pin 1. but no matter how i code it, the pins seam tied..

I tell it the Caps LED is pin 0 and the Num LED is pin 1.. i use:
"If (DigiKeyboard.getLEDs() && CAPS_LOCK) then set Caps LED to high.. change for if NUM_Lock then set Num LED high.. but then.. it turns on the Caps LED for both Caps and Num locks.. if i change it to pins 1 and 2, it'll turn on the board LED (so i guess that's wired to Pin 1) BUT it'll turn that on for BOTH Key presses.. Now, if i have to use pins 0 and 2 for the two LEDS, that's fine and understandable if the internal is interfearing, but I can't understand why the LED check is passing true for BOTH cases instead of only one..

The code is just this:
`#include "DigiKeyboard.h"
//LED Pins
#define LED_CAPS 1
#define LED_NUM 2
void setup()
{
  pinMode(LED_CAPS, OUTPUT);
 pinMode(LED_NUM, OUTPUT);
}
void loop()
{
  DigiKeyboard.delay(100);
  if (DigiKeyboard.getLEDs() && CAPS_LOCK)
  {
    digitalWrite(LED_CAPS, HIGH);
  } else {
    digitalWrite(LED_CAPS, LOW);
  }
  if (DigiKeyboard.getLEDs() && NUM_LOCK)
  {
    digitalWrite(LED_NUM, HIGH);
  } else {
    digitalWrite(LED_NUM, LOW);
  }
}`

trying to look though the code.. is the LED update to the DigiKeyboard library a bit.. lacking to NOT be able to tell the difference?

**Comment by ArminJo** [comment, marked as answer]
You know the difference between bitwise and logical boolean expressions?

**Reply by Tempy111** [reply]
ah.. damn.. point.. bitwise checks both sides where logical only checks the first (hand hand) side.. so logical (&&) would see that the getLEDs is true and not care about the second value..

too rusty on c ¬_¬ even though do a number of bits in C deviatives.. 
Thanks ^_^
